### ISSUE TITLE
Accessing Environment Variables from SQL queries

### ISSUE THREAD

**Issue by JoeHogan** [hasura/graphql-engine#3737]
I want to use Hasura not only to validate my JWTs, but also to generate them. I can configure postgres to generate JWTs but i need to pass the secret to the postgres function... I have already passed that same secret to hasura at startup (via the HASURA_GRAPHQL_JWT_SECRET environment variable), so I'd like to access the secret from the environment variable in my postgres function instead of hardcoding it. Is there any way to do this?

**Comment by JoeHogan** [comment]
for reference, im basically doing this: https://github.com/PostgREST/postgrest-docs/blob/master/auth.rst#client-auth

but in this process, they recommend: 

`-- run this once
ALTER DATABASE mydb SET "app.jwt_secret" TO 'reallyreallyreallyreallyverysafe';

-- then all functions can refer to app.jwt_secret
SELECT sign(
  row_to_json(r), current_setting('app.jwt_secret')
) AS token
FROM ...`

But again... I've already passed that secret to hasura at startup, so this is redundant if i could otherwise access it.

**Comment by JoeHogan** [comment]
i solved my own problem by adding a custom setting to postgres in my docker compose script:

`command: postgres -D /var/lib/postgresql/data -c 'supersecret'`

then accessing in postgres:

`current_setting('app.jwt_secret')`

**Comment by dminkovsky** [comment]
I was trying to accomplish something like this via the solution in https://dba.stackexchange.com/questions/52235/how-can-i-use-an-environment-variable-in-a-postgres-function, i.e. setting `-c ...` with `PGOPTIONS`. However, for some reason, I could access the resulting setting with from `psql -U postgres` but not from Hasura queries!

I got it working by directly appending the `-c ...` to `command` as mentioned by Joe above. No idea it didn't work the first way.

**Comment by jomarquez** [comment]
Amazon RDS for PostgreSQL does not allow you to add new config parameter in AWS RDS.

using `command: postgres -D /var/lib/postgresql/data -c 'app.jwt_secret=supersecret'` will not work here.

I went around this issue with setting our env the following way:
`ALTER database _db_name_ SET "_schema_._my_variable_" to '_variable_value_';`

Then restarting the hasura container:
docker-compose restart hasura

**Comment by Directrix-Digital** [comment]
I had the same challenge and found that your postgres command was the best solution I could get to work, However, instead of hardcoding the secret, I used docker compose and put the secret in an .env file, and referenced that in the command e.g.:
postgres -D /var/lib/postgresql/data -c 'app.jwt_secret=${JWT_SECRET}'
IMPORTANT if you inspect the postgres (v18) database container, the secret is visible in plaintext. But I haven't found a better way yet. I tried postgREST 14 configuration variables, but that didn't add the secret to the database.
